src/views/SystemList/Table.tsx

```diff
@@ -43,22 +50,83 @@ import {
 
 import './index.scss';
 
+type SystemTableType = 'all-systems' | 'my-systems' | 'bookmarked-systems';
+
 type TableProps = {
-  systems: CedarSystem[];
-  savedBookmarks: CedarSystemBookmark[];
-  refetchBookmarks: () => any;
+  systems?: CedarSystem[];
+  savedBookmarks?: CedarSystemBookmark[];
+  refetchBookmarks?: () => any;
+  defaultPageSize?: number;
+  isMySystems?: boolean;
 };
 
 export const Table = ({
-  systems,
-  savedBookmarks,
-  refetchBookmarks
+  systems = [],
+  savedBookmarks = [],
+  refetchBookmarks = () => null,
+  defaultPageSize = 10,
+  isMySystems
 }: TableProps) => {
   const { t } = useTranslation('systemProfile');
 
+  const location = useLocation();
+  const history = useHistory();
+
+  const params = useMemo(() => {
+    return new URLSearchParams(location.search);
+  }, [location.search]);
+  const tableType = params.get('table-type') as SystemTableType;
+
+  const [systemTableType, setSystemTableType] = useState<SystemTableType>(
+    tableType || 'all-systems'
+  );
+
+  const { loading, data: mySystems } = useQuery<GetMyCedarSystemsType>(
+    GetMyCedarSystemsQuery
+  );
+
   const [createMutate] = useMutation(CreateCedarSystemBookmarkQuery);
   const [deleteMutate] = useMutation(DeleteCedarSystemBookmarkQuery);
 
+  // Sets the systemTableType state to the query param, defaults to all-systems if no param present
+  // If the query param changes, update the component state
+  useEffect(() => {
+    if (!tableType && !isMySystems) {
+      history.replace({
+        search: 'table-type=all-systems'
+      });
+    }
+
+    if (isMySystems) {
+      setSystemTableType('my-systems');
+    } else {
+      setSystemTableType(tableType);
+    }
```

Comment:
nitpick: would the below work?
```suggestion
    setSystemTableType(isMySystems ? 'my-systems' : tableType);
```
because i believe this can fail to compile should the `else` condition be removed/changed later